app/ethos/views.py

```diff
@@ -0,0 +1,174 @@
+# -*- coding: utf-8 -*-
+'''
+    Copyright (C) 2017 Gitcoin Core
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+'''
+
+import json
+
+from django.conf import settings
+from django.http import Http404, JsonResponse
+from django.template.response import TemplateResponse
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+from django.views.decorators.csrf import csrf_exempt
+
+import twitter
+from ethos.models import Hop, ShortCode
+from gas.utils import recommend_min_gas_price_to_confirm_in_time
+from ratelimit.decorators import ratelimit
+from retail.helpers import get_ip
+from web3 import HTTPProvider, Web3
+
+w3 = Web3(HTTPProvider(settings.WEB3_HTTP_PROVIDER))
+
+
+@csrf_exempt
+@ratelimit(key='ip', rate='5/m', method=ratelimit.UNSAFE, block=True)
+def redeem_coin(request, shortcode):
+    print(shortcode)
+
+    if request.body:
+        status = 'OK'
+
+        body_unicode = request.body.decode('utf-8')
+        body = json.loads(body_unicode)
+
+        try:
+            twitter_api = get_twitter_api()
+
+            if not twitter_api:
+                raise Exception(_('Error while setting up Twitter API'))
+
+            address = body['address']
+            username = body['username']
+
+            twitter_api.GetUser(screen_name=username)
+
+            profile_pic = 'https://twitter.com/{}/profile_image?size=original'.format(username)
```

Comment:
Can you switch this to use literal string interpolation (f-strings)?

`profile_pic = f'https://twitter.com/{username}/profile_image?size=original'`